.githooks/commit-msg

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+declare -A components
+components=( \
+	["tools"]="tools" \
+	["s2i"]="tools/s2i-dpdk" \
+	["features"]="feature-configs" \
+	["ztp"]="ztp","go.mod","go.sum" \
+	["infra"]="Makefile",".githooks/" \
+	["vendor"]="vendor","cnf-tests","ztp" \
+	["cnf-tests"]="cnf-tests","go.mod","go.sum"
+)
+
+exceptions=( README.md )
+
+if ! [[ "$(head -1 $1)t sh" == *":"* ]]; then
+	echo ERROR: commit msg subject must include component name
+	exit 4
+fi
+
+msg_prefix=$(head -1 $1 | cut -d':' -f 1)
+if ! [ ${components[$msg_prefix]+exist} ]; then
+	printf -v components_string '%s/' "${!components[@]}"
+	echo "ERROR: must use authorized component (${components_string%/})"
+	exit 8
+fi
+
+for filename in $(git diff --name-only --cached); do
+	if [[ ! " ${exceptions[@]} " =~ " $filename " ]]; then
+		is_prefix_found=false
+		IFS=', ' read -r -a prefixes <<< "${components[$msg_prefix]}"
+		for prefix in "${prefixes[@]}"; do
+			if [[ "$filename" == "$prefix"* ]]; then
+				is_prefix_found=true
+				break
+			fi
+		done
+		if [ "$is_prefix_found" = false ]; then
+			echo ERROR: $filename is not part of $msg_prefix component
```

Comment:
can you add a dumb proof message?

Something like for file xxx your commit message must start with yyy , please change your commit message to zzz

Reply:
well, at least it's telling you what's wrong..
I'll try to add that logic

Reply:
there might be multiple matches

Reply:
done